### PR TITLE
Updates to package.json for 1.0.0 release

### PR DESCRIPTION
**Pull request by DMickens**

### CODE REVIEW

**Review by jonathanl-bq — APPROVED** [review]

File: packages/vertica-nodejs/package.json [hunk 1]
```diff
@@ -49,6 +49,6 @@
   ],
   "license": "Apache-2.0",
   "engines": {
-    "node": ">= 8.0.0"
+    "node": ">= 14.0.0"
```
Comment (williammatherjones) [inline packages/vertica-nodejs/package.json:52]: Why was this changed here?   I checked the spec and we did decided to support node 14 and higher.   I'm okay with the change, but want to double check to make absolutely certain this change won't cause problems.

File: packages/v-pool/package.json [hunk 2]
```diff
@@ -32,10 +32,9 @@
     "co": "4.6.0",
     "expect.js": "0.3.1",
     "lodash": "^4.17.11",
-    "mocha": "^7.1.2",
-    "pg-cursor": "^1.3.0"
```
Comment (williammatherjones) [inline packages/v-pool/package.json:36]: Another surprising change given the goal was just to change the version numbers.   Can you explain?   I suspect this is an unnecessary dependency, but want to make sure we discuss.

Comment (DMickens) [reply packages/v-pool/package.json:36]: pg-cursor was just another dependency that we have never been using but somehow slipped through during initial cleanup

File: packages/vertica-nodejs/package.json (hunk 1 above)
Comment (DMickens) [reply packages/vertica-nodejs/package.json:52]: The 'engines' specification in package.json is just a place to specify the version of node that our stuff works on. It has no impact unless you also have the engineStrict flag set to true. As it is with the change I've just made installing with npm if your node version is < 14 will give the user a warning, but will not keep them from installing. If the engine strict flag was set to true it would give them an error and say it's not compatible

This is just another way to say we think you should use node 14+, but we won't force you to
